src/org/zaproxy/zap/extension/fuzz/payloads/generator/RegexPayloadGenerator.java

```diff
@@ -215,6 +228,10 @@ public boolean hasNext() {
         @Override
         public DefaultPayload next() {
             count++;
+            if(randomOrder){
+                return new DefaultPayload(generex.random());
+            }
+
             return new DefaultPayload(iterator.next());
```

Comment:
`iterator.next()` needs to be called always, to ensure `hasNext` returns the proper value.

Reply:
For the record, when in random the intended behaviour is to generate payloads until the specified max, so it no longer needs to call `.next()` always (`hasNext` was updated accordingly).